jax/_src/numpy/lax_numpy.py

```diff
@@ -3530,10 +3530,13 @@ def msort(a):
 
 @_wraps(np.partition, lax_description="""
 The jax version requires the ``kth`` argument to be a static integer rather than
-a general array. This is implemented via two calls to :func:`jax.lax.top_k`.
+a general array. This is implemented via two calls to :func:`jax.lax.top_k`. If
+you're only accessing the top or bottom k values of the output, it may be more
+efficient to call :func:`jax.lax.top_k` directly.
 """)
 @partial(jit, static_argnames=['kth', 'axis'])
 def partition(a: ArrayLike, kth: int, axis: int = -1) -> Array:
+  # TODO(jakevdp): handle NaN values like numpy.
```

Comment:
Should we document this difference in behavior?

Reply:
Good idea - done!